griptape/structures/structure.py

```diff
@@ -194,3 +197,27 @@ def run(self, *args) -> Structure:
 
     @abstractmethod
     def try_run(self, *args) -> Structure: ...
+
+    def to_dict(self) -> dict:
+        """Converts attributes with metadata={"serializable": True} into a dictionary."""
+        serialized_data = {}
+        for _field in fields(self.__class__):
+            if _field.metadata.get("serializable", False):
+                value = getattr(self, _field.name)
+                serialized_data[_field.name] = self._serialize(value)
+
+        return serialized_data
+
+    def _serialize(self, value: Any) -> Any:
+        if hasattr(value, "to_dict"):  # Check if the object has a to_dict method
+            return value.to_dict()
+        elif isinstance(value, (list, tuple)):  # Handle lists or tuples
+            return [self._serialize(item) for item in value]
+        elif isinstance(value, dict):  # Handle dictionaries
+            return {key: self._serialize(val) for key, val in value.items()}
+        elif isinstance(value, Enum):  # Handle Enum types
+            return value.name  # Return the Enum's name as a string
+        elif callable(value):  # Handle functions or methods
+            return "default_function"
+        else:
+            return repr(value)  # Fallback for other non-serializable objects
```

Comment:
This functionality is provided by `SerializableMixin`, there is no need for a new method.